[extension/ackextension] release ackextension in alpha stability

zpzhuSplunk opened the pull request:
**Description:** <Describe what has changed.>
release ackextension in alpha stability so other components can start using it. 

**Link to tracking Issue:** <Issue number if applicable>
https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/26376

**Testing:** <Describe what testing was performed and which tests were added.>

**Documentation:** <Describe the documentation added.>

zpzhuSplunk committed:
[extension/ackextension] release ackextension in alpha stability

zpzhuSplunk committed:
add chloggen

zpzhuSplunk committed:
test

zpzhuSplunk committed:
test

zpzhuSplunk committed:
test

zpzhuSplunk committed:
Merge branch 'main' into release-ack-extension

zpzhuSplunk committed:
test

zpzhuSplunk committed:
Merge branch 'main' into release-ack-extension

zpzhuSplunk committed:
test

zpzhuSplunk committed:
test

crobert-1 commented on extension/ackextension/README.md:
```diff
@@ -1,11 +1,11 @@
-| [Code Owners](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/CONTRIBUTING.md#becoming-a-code-owner)    | [@zpzhuSplunk](https://www.github.com/zpzhuSplunk), [@splunkericl](https://www.github.com/splunkericl) |
+| [Code Owners](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/CONTRIBUTING.md#becoming-a-code-owner)    | [@zpzhuSplunk](https://www.github.com/zpzhuSplunk), [@splunkericl](https://www.github.com/splunkericl)                                                                                                                                                                  ...
 
 [development]: https://github.com/open-telemetry/opentelemetry-collector#development
```
Something is going wrong with what's being generated here, this should be updated to show `alpha` instead. Can you maybe try removing the whole autogenerated section and running the generate command again?

crobert-1 commented on .chloggen/release-ack-extension.yaml:
```diff
@@ -0,0 +1,27 @@
+component: ackextension
+
+# A brief description of the change.  Surround your text with quotes ("") if it needs to start with a backtick (`).
+note: "releasing the ackextension in alpha stability"
```
```suggestion
note: "Promote the to `alpha` stability"
```
Just a nit, formatted it to be a bit more similar to other release notes of the same kind.

crobert-1 commented on .chloggen/release-ack-extension.yaml:
```diff
@@ -0,0 +1,27 @@
+# Use this changelog template to create an entry for release notes.
+
+# One of 'breaking', 'deprecation', 'new_component', 'enhancement', 'bug_fix'
+change_type: enhancement
```
```suggestion
change_type: new_component
```
It might make sense to make this `new_component` to show that it's now available for use.

crobert-1 commented on .chloggen/release-ack-extension.yaml:
```diff
@@ -0,0 +1,27 @@
+# Include 'user' if the change is relevant to end users.
+# Include 'api' if there is a change to a library API.
+# Default: '[user]'
+change_logs: [api]
```
```suggestion
change_logs: [user]
```
This is relevant to end users, so I think it should be `user` here.

crobert-1 reviewed (commented):
`v0.98.0` was just released, so version numbers here should be updated to match 👍

zpzhuSplunk committed:
address MR comments

zpzhuSplunk committed:
Merge branch 'main' into release-ack-extension

crobert-1 commented on internal/components/components.go:
```diff
@@ -209,6 +210,7 @@ func Components() (otelcol.Factories, error) {
 		oidcauthextension.NewFactory(),
 		sigv4authextension.NewFactory(),
 		zpagesextension.NewFactory(),
+		ackextension.NewFactory(),
```
Nit: The extensions are in alphabetical order so far, this should be moved up to match existing order.

zpzhuSplunk committed:
test

zpzhuSplunk committed:
test

zpzhuSplunk committed:
test

zpzhuSplunk committed:
Merge branch 'main' into release-ack-extension

zpzhuSplunk committed:
test

zpzhuSplunk committed:
test

zpzhuSplunk committed:
test

crobert-1 reviewed (commented):
Failing CI/CD action:
```
Generated code is out of date, please run "make genotelcontribcol" and commit the changes in this PR.
```

zpzhuSplunk committed:
Merge branch 'main' into release-ack-extension

zpzhuSplunk committed:
test

zpzhuSplunk committed:
Merge branch 'main' into release-ack-extension

zpzhuSplunk committed:
test'

zpzhuSplunk committed:
test

zpzhuSplunk committed:
test

zpzhuSplunk committed:
test

zpzhuSplunk committed:
test

zpzhuSplunk committed:
test

zpzhuSplunk committed:
Merge branch 'main' into release-ack-extension

zpzhuSplunk committed:
Merge branch 'main' into release-ack-extension

zpzhuSplunk committed:
test

zpzhuSplunk committed:
test

zpzhuSplunk committed:
Merge branch 'main' into release-ack-extension

zpzhuSplunk committed:
Merge branch 'main' into release-ack-extension

crobert-1 reviewed (commented):
Looks like at least one reference is missing. From the [Adding New Components section of the CONTRIBUTING.md](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/CONTRIBUTING.md#adding-new-components) doc:
```
Last PR should mark the new component as Alpha stability and add it to the cmd/otelcontribcol binary by updating the cmd/otelcontribcol/components.go file.
```

zpzhuSplunk commented:
> Looks like at least one reference is missing. From the [Adding New Components section of the CONTRIBUTING.md](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/CONTRIBUTING.md#adding-new-components) doc:
> 
> ```
> Last PR should mark the new component as Alpha stability and add it to the cmd/otelcontribcol binary by updating the cmd/otelcontribcol/components.go file.
> ```

originally I modified this file, but the commands I had to execute would remove what I added. On line 1 in the file, it seems like this is generated elsewhere? https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/cmd/otelcontribcol/components.go#L1

zpzhuSplunk committed:
Merge branch 'main' into release-ack-extension

crobert-1 commented:
My apologies, you're right, it's not clear. It's being generated from [here](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/cmd/otelcontribcol/builder-config.yaml), so you'd actually want to update the builder config. I believe you'll also want to add the extension to the [extension tests here](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/cmd/otelcontribcol/extensions_test.go) as well.

zpzhuSplunk committed:
address PR comments

zpzhuSplunk committed:
test

zpzhuSplunk committed:
test

crobert-1 commented on cmd/otelcontribcol/builder-config.yaml:
I believe you also want to add the component to the [extensions block near the top of this file](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/6255f2b1656d986c16024870d19faabb684a86ae/cmd/otelcontribcol/builder-config.yaml#L8).

zpzhuSplunk replied:
Ok I added the component. `make checks` seemed to have resolved my component to version v0.0.0-00010101000000-000000000000. I assume this is normal given that the extension hasn't been released yet. Is this the right version to use?

crobert-1 replied:
I don't think it matters either way as the replace will simply override whatever is stated with the current contents of the relative dir below. However, could you test to see if CI passes with `v0.98.0`? (Same with other `v0.0...` reference in `cmd/otelcontribcol/go.mod`)

zpzhuSplunk commented:
Hi @dmitryax, minding taking a look?

zpzhuSplunk committed:
add extension

zpzhuSplunk committed:
test

crobert-1 commented on cmd/otelcontribcol/extensions_test.go:
```diff
@@ -20,6 +20,8 @@ import (
 	"go.opentelemetry.io/collector/extension/zpagesextension"
 
+	"github.com/open-telemetry/opentelemetry-collector-contrib/extension/ackextension"
+
```
```suggestion
```

zpzhuSplunk committed:
test

zpzhuSplunk committed:
Merge branch 'main' into release-ack-extension

zpzhuSplunk committed:
test

zpzhuSplunk committed:
test

zpzhuSplunk committed:
Merge branch 'main' into release-ack-extension

zpzhuSplunk committed:
Merge branch 'main' into release-ack-extension

zpzhuSplunk committed:
Merge branch 'main' into release-ack-extension

zpzhuSplunk committed:
test

zpzhuSplunk committed:
Merge branch 'main' into release-ack-extension

atoulme reviewed (approved):
LGTM
